### PR TITLE
Add nodes to document tree

### PR DESCRIPTION
**Pull request by elliotmjackson**
Pathing the way for upcoming getting started guides

1. Create new `Guides` category to house tour and how-to.
2. Create new `The Buf CLI` category to include cli features

no links change, only sidebar display.

### CODE REVIEW

File: sidebars.js [hunk 1]
```diff
@@ -6,86 +6,112 @@ const sidebars = {
...
-                    customProps: {
-                        badge: {
-                            label: "new",
-                            severity: "info"
-                        }
-                    }
+                    type: "category",
+                    label: "Tour",
+                    items: [
+                        "tour/introduction",
```
Comment (joshcarp) [inline sidebars.js:15]: Question: Would it be possible to change the `introduction` to `overview`. This way there's a nice symmetry between this and the buf cli reference docs which has `overview` as the category index page?

Comment (elliotmjackson) [reply sidebars.js:15]: out of scope, the tour is getting deleted

**Comment by joshcarp** [comment]
Do we need to make it so that old links aren't broken?

**Comment by elliotmjackson** [comment]
highlighted in the description of the PR "no links change, only sidebar display".

**Review by joshcarp — APPROVED** [review]

**Pull request merged**